perfkitbenchmarker/providers/ibmcloud/ibmcloud_disk.py

```diff
@@ -0,0 +1,199 @@
+                                                                      vm.name)
+      raise errors.Error(msg)
+
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['name'] = self.name
+    volcmd.flags['instanceid'] = vm.vmid
+    volcmd.flags['volume'] = self.vol_id
+    volcmd.flags['delete'] = True
+
+    # wait till volume is PROVISIONED before attach
+    resp = json.loads(volcmd.ShowVolume())
+    status = resp['status']
+    endtime = time.time() + 300
+    while status != AVAILABLE and time.time() < endtime:
+      time.sleep(2)
+      status = json.loads(volcmd.ShowVolume())['status']
+      logging.info('Checking volume status: %s', status)
+
+    if status != AVAILABLE:
+      logging.error('Failed to create a volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to provision a volume.')
+
+    resp = json.loads(volcmd.InstanceCreateVolume())
+    logging.info('Attached volume, volcmd.flags %s', volcmd.flags)
+    volume = resp['id']
+    logging.info('VDISK ID: %s, vdiskname: %s, instanceid: %s', volume, self.name, vm.vmid)
+    self.attached_vdisk_URI = volume
+    self.attached_vm = vm
+    self._WaitForVolumeAttachment(vm)
+    self._GetDeviceFromVDisk(vm)
+
+  @vm_util.Retry(poll_interval=1, max_retries=-1, timeout=300, log_errors=False,
+                 retryable_exceptions=errors.Resource.RetryableCreationError)
+  def _WaitForVolumeAttachment(self, vm):
+    """Waits and checks until the volume status is attached """
+    if self.vol_id is None:
+      return
+    volcmd = ibm.IbmAPICommand(self)
+    volcmd.flags['instanceid'] = self.attached_vm.vmid
+    volcmd.flags['volume'] = self.vol_id
+    logging.info('Checking volume on instance %s, volume: %s', vm.vmid, self.vol_id)
+    resp = volcmd.InstanceShowVolume()
+    logging.info('Show volume resp: %s', resp)
+    status = None
+    if resp:
+      resp = json.loads(resp)
+      status = resp['status'] if resp and 'status' in resp else 'unknown'
+
+    endtime = time.time() + 300
+    while status != ATTACHED and time.time() < endtime:
+      time.sleep(2)
+      resp = volcmd.InstanceShowVolume()
+      if resp:
+        resp = json.loads(resp)
+        status = resp['status'] if resp and 'status' in resp else 'unknown'
+        logging.info('Checking instance volume status: %s', status)
+
+    if status == ATTACHED:
+      logging.info('Remote volume %s has been attached to %s', self.name, vm.name)
+    else:
+      logging.error('Failed to attach the volume')
+      raise errors.Error('IBMCLOUD ERROR: failed to attach a volume.')
+
+  def _GetDeviceFromVDisk(self, vm):
+    """Gets device path for the volume """
+    cmd = 'fdisk -l'
+    endtime = time.time() + 120
+    self.device_path = None
+    while time.time() < endtime:
+      stdout, _ = self.attached_vm.RemoteCommand(cmd, should_log=True)
+      # parse for lines that contain "Disk /dev/vdd: 1000 GiB"
+      disks = re.findall(r'\Disk /.*/.*: \d+ GiB', stdout)
```

Comment:
Can you do something like this?
 disks = re.findall(r'\Disk (\S+): (\d+) GiB', stdout)
 for disk_path, disk_size in disks:
    ...

Reply:
yes thanks for the tip, that is much better, updated.